zerver/tests/test_user_groups.py

```diff
@@ -2396,6 +2399,9 @@ def check_removing_members_from_group(
         do_change_user_group_permission_setting(
             user_group, "can_manage_group", setting_group, acting_user=None
         )
+        do_change_user_group_permission_setting(
+            user_group, "can_add_members_group", setting_group, acting_user=None
+        )
```

Comment:
I don't understand what you're doing with these tests -- why are you setting both of these settings to the same value everywhere? I feel like a more natural test design would be to set one setting to NOBODY to test the other in isolation, and then do the opposite.

Reply:
Split into two tests, one for adding and one for removing members. For adding members, I'm not sure if we need to test `can_manage_group` for all possible cases while keeping `can_add_members_group` as nobody, the one block at the end of that test might be enough.

Reply:
I've added this as a fixup commit on top since one of the commits was removing some tests, and the fixup commit was easier to do than splitting tests in the first commit and then changing those tests again in the commits after that. Let me know if rebasing would be preferred here.

Reply:
Yeah, one block for testing that is enough.